pkg/occlient/occlient_test.go

```diff
@@ -406,3 +408,883 @@ func Test_parseImageName(t *testing.T) {
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								"deploymentconfig": "wildfly",
+							},
+						},
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Image: "wildfly:latest",
+									Name:  "wildfly",
+								},
+							},
+						},
+					},
+				},
+			},
+			createdPVC: corev1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: fmt.Sprintf("%s-s2idata", "wildfly"),
+					Labels: map[string]string{
+						"app": "apptmp",
+						"app.kubernetes.io/component-name": "wildfly",
+						"app.kubernetes.io/component-type": "wildfly",
+						"app.kubernetes.io/name":           "apptmp",
+					},
+				},
+				Spec: corev1.PersistentVolumeClaimSpec{
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: errQuantity,
+						},
+					},
+					AccessModes: []corev1.PersistentVolumeAccessMode{
+						corev1.ReadWriteOnce,
+					},
+				},
+			},
+
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fkclient, fkclientset := FakeNew()
+			fkclientset.AppsClientset.PrependReactor("get", "deploymentconfigs", func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+				if tt.dcName != tt.existingDc.Name {
+					return true, nil, fmt.Errorf("got different dc")
+				}
+				return true, &tt.existingDc, nil
+			})
+
+			fkclientset.AppsClientset.PrependReactor("update", "deploymentconfigs", func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+				dc := action.(ktesting.UpdateAction).GetObject().(*appsv1.DeploymentConfig)
+				if dc.Name != tt.existingDc.Name {
+					return true, nil, fmt.Errorf("got different dc")
+				}
+				return true, dc, nil
+			})
+
+			fkclientset.Kubernetes.PrependReactor("create", "persistentvolumeclaims", func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+				createdPvc := action.(ktesting.CreateAction).GetObject().(*corev1.PersistentVolumeClaim)
+				if createdPvc.ObjectMeta.Name != tt.createdPVC.ObjectMeta.Name {
+					return true, nil, fmt.Errorf("got a different pvc name")
+				}
+				if !reflect.DeepEqual(createdPvc.Spec, tt.createdPVC.Spec) {
+					return true, nil, fmt.Errorf("got a different pvc spec")
+				}
+				return true, &tt.createdPVC, nil
+			})
+
+			err := fkclient.SetupForSupervisor(tt.dcName, tt.projectName, tt.annotations, tt.labels)
+
+			if err == nil && !tt.wantErr {
+				// Check for validating actions performed
+				if (len(fkclientset.AppsClientset.Actions()) != 2) && (tt.wantErr != true) {
+					t.Errorf("expected 2 action in UpdateDeploymentConfig got: %v", fkclientset.AppsClientset.Actions())
+				}
+
+				// Check for validating actions performed
+				if (len(fkclientset.Kubernetes.Actions()) != 1) && (tt.wantErr != true) {
+					t.Errorf("expected 1 action in CreatePVC got: %v", fkclientset.AppsClientset.Actions())
+				}
+
```

Comment:
The checks below should be run only if the `err` is `nil`.

Reply:
But the prependReactors can return error too

Reply:
hmm, that would be a case of passing incorrect values. Anyways, no harm keeping it.